wifi_logger.c

```diff
@@ -237,29 +264,53 @@ void wifi_logger()
 #ifdef CONFIG_TRANSPORT_PROTOCOL_TCP
 void wifi_logger()
 {
-    struct tcp_network_data* handle = malloc(sizeof(struct network_data));
-    tcp_network_manager(handle);
+	char* log_message;
+	while(1) // To keep trying over and over
+	{
+		while(1) // Try creating and connecting to socket
+		{
+			struct tcp_network_data* handle = malloc(sizeof(struct network_data));
+			if (false == tcp_network_manager(handle))
+			{
+				// Slow down while retrying.
+				vTaskDelay(2000 / portTICK_PERIOD_MS);
+				break;
```

Comment:
You need to free(handle) before breaking out

Reply:
already done inside the tcp_network_manager func when it fails, it calls tcp_close_network_manager.

Reply:
My bad ! Just read it